cpu/atmega_common/periph/cpuid.c

```diff
@@ -44,24 +37,12 @@
  */
 void cpuid_get(void *id)
 {
-    uint8_t  signature_0 = boot_signature_byte_get(0x00);
-    uint8_t  signature_1 = boot_signature_byte_get(0x02);
-    uint8_t  signature_2 = boot_signature_byte_get(0x04);
-
-    uint8_t  usr_sign_0 = boot_signature_byte_get(0x0100);
-    uint8_t  usr_sign_1 = boot_signature_byte_get(0x0102);
-
-    uint8_t addr[CPUID_LEN] = {
-            signature_0,        /* 0x1E Atmel manufacturer ID */
-            signature_1,        /* 0xA8 Part Number high byte */
-            signature_2,        /* 0x02 Part Number low byte */
-            MAN_ID_0,           /* 0x1F Atmel JEDEC manufacturer ID */
-            PART_NUM,           /* 0x94 PART_NUM  Identification */
-            VERSION_NUM,        /* 0x02 VERSION_NUM  Identification */
```

Comment:
No idea about `atmega`, but from their names don't these fields also provide uniqueness?

Reply:
No, they identify the Manufacturer/Part and are the same for each device.
`MAN_ID_0` is always `0x1F` for Atmel, `PART_NUM` will be `0x94` for ATmega256RFR2  and `VERSION_NUM` is 2 for revision 2 of the chip.